openshift/ci-operator/knative-images/cli-artifacts/Dockerfile

```diff
@@ -1,33 +1,7 @@
 # This is not generated Dockerfile, yet!
-ARG GO_BUILDER=registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.22-openshift-4.17
-ARG GO_RUNTIME=registry.access.redhat.com/ubi8/ubi-minimal
+ARG CLI_ARTIFACTS=brew.registry.redhat.io/rh-osbs/openshift-serverless-1-kn-cli-artifacts-rhel8@sha256:c0376504edae12e7af6b6cce02ac53b6b374f5058ebefcbdbd32628b551bb0c4
```

Comment:
The default value for CLI_ARTIFACTS is for OpenShift CI. We need a public image there. Something like we have for must-gather here: https://github.com/openshift-knative/serverless-operator/blob/main/must-gather/Dockerfile#L2
We will then override it in Konflux with this new image through ARGS

Reply:
We'll add it [here](https://github.com/openshift-knative/hack/blob/main/config/serverless-operator.yaml#L8) like this:
```
        - name: JAVA_RUNTIME
          pullSpec: brew.registry.redhat.io/rh-osbs/openshift-serverless-1-kn-cli-artifacts-rhel8@sha256:c0376504edae12e7af6b6cce02ac53b6b374f5058ebefcbdbd32628b551bb0c4
```

Reply:
For the public ref image this can always be `registry.redhat.io` of course.

Reply:
Discussed on slack.